### PR TITLE
Update the showcase to backstage 1.17

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

- Update backstage deps
- Update non backstage deps
- Pin all deps

https://backstage.github.io/upgrade-helper/?from=1.16.0&to=1.17.0

## Which issue(s) does this PR fix

- Fixes https://github.com/janus-idp/backstage-showcase/issues/434
- Blocked by https://github.com/janus-idp/backstage-plugins/pull/650
- blocked by https://github.com/janus-idp/backstage-plugins/issues/655

## PR acceptance criteria

Please make sure that the following steps are complete:

- GitHub Actions are completed and successful
- Unit Tests are updated and passing
- E2E Tests are updated and passing
- Documentation is updated if necessary
- Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: .changeset/beige-jokes-add.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+'backend': major
+'app': major
+---
+
+Update the proxy object to include the endpoint property in `app-config.yaml`
```
Comment (christophe-f) [inline .changeset/beige-jokes-add.md:6]: @schultzp2020 @tumido , this is a breaking change. Should we mention this here, or it's better with the Helm Chart release as customers will have to update their config files?

File: package.json [hunk 2]
```diff
@@ -30,18 +30,17 @@
...
-    "@spotify/prettier-config": "^12.0.0",
-    "cross-var": "^1.1.0",
-    "eslint-plugin-import": "^2.27.5",
-    "husky": "^8.0.3",
-    "prettier": "^2.3.2",
-    "turbo": "^1.9.3",
-    "typescript": "^5.0.4"
+    "@backstage/cli": "0.22.12",
+    "@changesets/cli": "2.26.2",
+    "@spotify/prettier-config": "15.0.0",
```
Comment (christophe-f) [inline package.json:35]: Backstage 1.17 seems to still be using 12.0.0

File: package.json [hunk 3]
```diff
@@ -30,18 +30,17 @@
...
+    "cross-var": "1.1.0",
```
Comment (christophe-f) [inline package.json:36]: Concurrently dependency is missing

File: packages/app/public/index.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (christophe-f) [inline packages/app/public/index.html:1]: Was it missed from a previous update?

**Review by christophe-f — APPROVED** [review]
LGTM

File: package.json (hunk 2 above)
Comment (schultzp2020) [reply package.json:35]: I think we should stick with the latest version. From what I have read in their [changelog](https://github.com/spotify/web-scripts/releases), there doesn't seem to be any rule changes.

File: package.json (hunk 3 above)
Comment (schultzp2020) [reply package.json:36]: Concurrently is not needed because of `turbo`.

File: packages/app/public/index.html (hunk 4 above)
Comment (schultzp2020) [reply packages/app/public/index.html:1]: I think this changed when `prettier` was updated to `3.y.z`.

**Pull request merged**